### ISSUE TITLE
Question about unmatched points in Eq. (10)? 

### ISSUE THREAD

**Issue by GabbySuwichaya** [magicleap/SuperGluePretrainedNetwork#97]
Hi! Thank you for the great work. I enjoy reading your paper. 

Here, I am would like to ask for your help with the unmatched points in the set I and J of Eq.(10) .. 
 
From your explanation, I and J contains the points (i,j) that are unmatched if they do not have any re-projection in their vicinity... 

Do you mean that, for example, if we are matching  4 keypoints with another 4 keypoints, and the set of the ground truth match set M contains (1,1), (2,2), (3,2) and (3,4)...  

So, the set I will contain  {(4,1), (4,2), (4,3), (4,4)} and the set J will contain {(1,3), (2,3), (3,3), (4,3)} Are these correct ? Could you please confirm ? 

What about {(1,2),(1,4),(2,1),(2,4), and (3,1) } ? are they also in the set ?

**Comment by GabbySuwichaya** [comment]
I have written my own code to train SuperGlue.

What I really would like to ask is how to solve the problem of **high false positives**? 

-  In training, I have set the following: 

```
- torch.optim.Adam with learning rate of 1e-4 and exponential decay of 0.99999992
- Sinkhorn matching iteration = 50
```

-  Loss function:
I have used the following way to construction Eq. (10). Maybe it would be nice if you could confirm if it is a correct way of interpreting the probability corresponding to set I and J. 

```
self.loss = nn.BCEWithLogitsLoss() 
P, PI, PJ = Z[:b,:m,:n], Z[:b,m,:], Z[:b,:,n]
neg_edge_lable_I    = torch.zeros_like(PI, dtype=bool)  
neg_edge_lable_J   = torch.zeros_like(PJ, dtype=bool)  

P_  = torch.cat([P, PI, PJ]) 
lable_ = torch.cat([pos_edge_lable, neg_edge_lable_I, neg_edge_lable_J])  

loss += self.loss(P_, lable_.float())   
```


Here are the stats that I collected at the training time, where I evaluated the MMA. Also, I have measured the Euclidian error of the matched points based on the homography, and the number of matches during training.  

| MMA        | Euclidian Error |  Number of matches |
| ------------- | ------------- |  ------------- |
| ![Loss-MMA-3px-epoch-010-idx-002000](https://user-images.githubusercontent.com/49115268/129516088-53e9516f-81e2-4dd5-a321-a9db4762f7e0.png)  |  ![Loss-MME-all-epoch-010-idx-002000](https://user-images.githubusercontent.com/49115268/129516274-c9de8ef6-1540-4d06-a106-3c726bdedd97.png) | ![Loss-NumMatches-epoch-010-idx-002000](https://user-images.githubusercontent.com/49115268/129516132-fc190935-5aa5-4e3d-83ea-99357100ff19.png) |
